plugins/trivy/cmd/trivy/main.go

```diff
@@ -174,3 +105,93 @@ func getNewestVersions(versionsChan chan newestVersions, ctx context.Context, im
 		versions: versions,
 	}
 }
+
+func getShaFromID(id string) string {
+	if len(strings.Split(id, "@")) > 1 {
+		return strings.Split(id, "@")[1]
+	}
+	return id
+}
+
+func getImagesToScan(images []models.Image, lastReportImages []models.ImageDetailsWithRefs) []models.Image {
+	imagesToScan := make([]models.Image, 0)
+	for _, image := range images {
+		found := false
+
+		for _, report := range lastReportImages {
+			reportSha := getShaFromID(report.ID)
+			imageSha := getShaFromID(image.ID)
+			if report.Name == image.Name && reportSha == imageSha {
+				found = true
+				break
+			}
+		}
+		if !found {
+			imagesToScan = append(imagesToScan, image)
```

Comment:
I think we need `if len(imagesToScan) < numberToScan {`

Reply:
I was doing that after the call, but I believe I had a bug. Glad about this comment